tests/test_sdk_client.py

```diff
@@ -114,5 +120,46 @@ def mock_get_password():
 
 
 def test_validate_connection_uses_given_credentials(mock_sdk_factory):
-    assert validate_connection("Authority", "Test", "Password")
-    mock_sdk_factory.assert_called_once_with("Authority", "Test", "Password")
+    assert validate_connection("Authority", "Test", "Password", None)
+    mock_sdk_factory.assert_called_once_with("Authority", "Test", "Password", totp=None)
+
+
+def test_validate_connection_when_mfa_required_exception_raised_prompts_for_totp(
+    mocker, monkeypatch, mock_sdk_factory, capsys
+):
+    monkeypatch.setattr("sys.stdin", StringIO("101010"))
+    response = mocker.MagicMock(spec=Response)
+    mock_sdk_factory.side_effect = [
+        Py42MFARequiredError(HTTPError(response=response)),
+        None,
+    ]
+    validate_connection("Authority", "Test", "Password", None)
+    output = capsys.readouterr()
+    assert "Multi-factor authentication required. Enter TOTP:" in output.out
+
+
+def test_validate_connection_when_mfa_token_invalid_raises_expected_cli_error(
+    mocker, mock_sdk_factory
+):
+    response = mocker.MagicMock(spec=Response)
+    response.text = '{"data":null,"error":[{"primaryErrorKey":"INVALID_TIME_BASED_ONE_TIME_PASSWORD","otherErrors":null}],"warnings":null}'
+    mock_sdk_factory.side_effect = Py42UnauthorizedError(HTTPError(response=response))
+    with pytest.raises(Code42CLIError) as err:
+        validate_connection("Authority", "Test", "Password", "1234")
+    assert str(err.value) == "Invalid TOTP token for user Test."
+
+
+def test_totp_option_when_passed_is_passed_to_sdk_initialization(
+    mocker, profile, runner
+):
+    mock_py42 = mocker.patch("code42cli.sdk_client.py42.sdk.from_local_account")
+    cli_state = CLIState()
+    totp = "1234"
+    profile.authority_url = "test.com"
```

Comment:
`example.com` is generally better.

I once accidentally made a network request to something like `test.com`